src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -206,16 +206,16 @@ public class ADTaskManager {
 
     public ADTaskManager(
         Settings settings,
-        ClusterService clusterService,
-        Client client,
+        SDKClusterService clusterService,
+        SDKRestClient sdkRestClient,
         NamedXContentRegistry xContentRegistry,
```

Comment:
Can we replace this with `SDKNamedXContentRegistry`?

Reply:
What @owaiskazi19 said. The `SDKNamedXContentRegistry` is injected at instantiation. Use its getter when you need to access the "current" registry interanally in the classes.

Reply:
Carry over from my earlier comment about this class in `createComponents()`.  Elsewhere in this class add `getRegistry()` to the usages to this field.